plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -242,11 +245,36 @@ func (s *SQLServer) Gather(acc telegraf.Accumulator) error {
 }
 
 func (s *SQLServer) gatherServer(server string, query Query, acc telegraf.Accumulator) error {
-	// deferred opening
-	conn, err := sql.Open("mssql", server)
-	if err != nil {
-		return err
+	// deferred connection opening
+	var conn *sql.DB
+
+	if strings.Contains(strings.ToLower(server), strings.ToLower("password=")) {
+		// when password is provided in connection string, use SQL auth
+
+		var err error
+		conn, err = sql.Open("mssql", server)
+
+		if err != nil {
+			return err
+		}
+	} else {
+		// when password is not provided in connection string, use AAD auth with an MSI token
+
+		tokenProvider, err := getMSITokenProvider()
```

Comment:
For SQL VM, should we throw an exception and fail sooner here? SQL VM does not support AAD auth

Reply:
Added a conditional check for DatabaseType = SQLServer i.e. SQL VM to error out if AAD auth is used